client/index.go

```diff
@@ -36,6 +36,18 @@ type IndexDescription struct {
 	Unique bool
 }
 
+// IndexCreateRequestDescription describes an index creation request.
+// It does not contain the ID, as it is not a valid field for the request body.
+// Instead it should be automatically generated.
+type IndexCreateRequestDescription struct {
```

Comment:
todo: The client interface should be updated to use this new struct.

I would suggest moving it to the file below and changing the `IndexDescription` parameter.

https://github.com/sourcenetwork/defradb/blob/develop/client/collection.go#L118

Reply:
nitpick: `IndexDescriptionCreateRequest` might be more accurate.

Reply:
I have addressed the variable name, that was straightforward.

I have also adjusted the code as Keenan has asked. This resulted in a lot of cascading (small) changes to functions, and function signatures, and in many changes to the integration tests to accommodate those changes. I would like to know if this is what was desired. I will mark for re-review, but I do know that Keenan is out of office this week.